src/ddmd/astbaremetal.d

```diff
@@ -0,0 +1,3248 @@
+            this.packages = packages;
+            this.id = id;
+            this.aliasId = aliasId;
+            this.isstatic = isstatic;
+            this.protection = Prot(PROTprivate);
+
+            if (aliasId)
+            {
+                // import [cstdio] = std.stdio;
+                this.ident = aliasId;
+            }
+            else if (packages && packages.dim)
+            {
+                // import [std].stdio;
+                this.ident = (*packages)[0];
+            }
+            else
+            {
+                // import [foo];
+                this.ident = id;
+            }
+        }
+        void addAlias(A, B)(A a, B b) {}
+    }
+
+    extern (C++) abstract class AttribDeclaration : Dsymbol
+    {
+        Dsymbols* decl;
+
+        final extern (D) this(Dsymbols *decl)
+        {
+            this.decl = decl;
+        }
+    }
+
+    extern (C++) final class StaticAssert : Dsymbol
+    {
+        Expression exp;
+        Expression msg;
+
+        final extern (D) this(Loc loc, Expression exp, Expression msg)
+        {
+            super(Id.empty);
+            this.loc = loc;
+            this.exp = exp;
+            this.msg = msg;
+        }
+    }
+
+    extern (C++) final class DebugSymbol : Dsymbol
+    {
+        uint level;
+
+        extern (D) this(Loc loc, Identifier ident)
+        {
+            super(ident);
+            this.loc = loc;
+        }
+        extern (D) this(Loc loc, uint level)
+        {
+            this.level = level;
+            this.loc = loc;
+        }
+    }
+
+    extern (C++) final class VersionSymbol : Dsymbol
+    {
+        uint level;
+
+        extern (D) this(Loc loc, Identifier ident)
+        {
+            super(ident);
+            this.loc = loc;
+        }
+        extern (D) this(Loc loc, uint level)
+        {
+            this.level = level;
+            this.loc = loc;
+        }
+    }
+
+    extern (C++) class VarDeclaration : Declaration
+    {
+        Type type;
+        Initializer _init;
+        StorageClass storage_class;
+        int ctfeAdrOnStack;
+        uint sequenceNumber;
+        __gshared uint nextSequenceNumber;
+
+        final extern (D) this(Loc loc, Type type, Identifier id, Initializer _init, StorageClass st = STCundefined)
+        {
+            super(id);
+            this.type = type;
+            this._init = _init;
+            this.loc = loc;
+            this.storage_class = storage_class;
+            sequenceNumber = ++nextSequenceNumber;
+            ctfeAdrOnStack = -1;
+        }
+    }
+
+    extern (C++) class FuncDeclaration : Declaration
+    {
+        Statement fbody;
+        Statement frequire;
+        Statement fensure;
+        Loc endloc;
+        Identifier outId;
+        StorageClass storage_class;
+        Type type;
+        bool inferRetType;
+        ForeachStatement fes;
+
+        final extern (D) this(Loc loc, Loc endloc, Identifier id, StorageClass storage_class, Type type)
```

Comment:
I saw that all (or most) constructors are declared as extern in the standard implementation. Since I pretty much copy pasted the constructors, I thought that the extern keyword should be kept